nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java

```diff
@@ -210,17 +218,23 @@ private void setConnectionFactoryProperties(ConfigurationContext context) {
             if (descriptor.isDynamic()) {
                 this.setProperty(propertyName, entry.getValue());
             } else {
-                if (propertyName.equals(BROKER)) {
-                    String brokerValue = context.getProperty(descriptor).evaluateAttributeExpressions().getValue();
-                    if (context.getProperty(CONNECTION_FACTORY_IMPL).evaluateAttributeExpressions().getValue().startsWith("org.apache.activemq")) {
+                if (descriptor == BROKER_URI) {
+                    String brokerValue = context.getProperty(BROKER_URI).evaluateAttributeExpressions().getValue();
+                    String connectionFactoryValue = context.getProperty(CONNECTION_FACTORY_IMPL).evaluateAttributeExpressions().getValue();
+                    if (connectionFactoryValue.startsWith("org.apache.activemq")) {
                         this.setProperty("brokerURL", brokerValue);
+                    } else if (connectionFactoryValue.startsWith("com.tibco.tibjms")) {
+                        this.setProperty("serverUrl", brokerValue);
                     } else {
+                        // Try to parse broker URI as colon separated host/port pair
                         String[] hostPort = brokerValue.split(":");
+                        // If broker URI indeed was colon separated host/port pair
                         if (hostPort.length == 2) {
                             this.setProperty("hostName", hostPort[0]);
                             this.setProperty("port", hostPort[1]);
-                        } else if (hostPort.length != 2) {
-                            this.setProperty("serverUrl", brokerValue); // for tibco
+                        } else if (connectionFactoryValue.startsWith("com.ibm.mq.jms")) {
+                            // Assuming IBM MQ style broker was specified, e.g. "myhost(1414)" and "myhost01(1414),myhost02(1414)"
+                            this.setProperty("connectionNameList", brokerValue);
```

Comment:
After reading some articles and API specs for connectionNameList property, just wondering if setting `connectionNameList` by a dynamic property work. Did you try before working on this PR?

What if both `host` and `port` pair and `connectionNameList` are set?

> Specifies the hosts to which the client will attempt to reconnect to after its connection is broken.
https://www.ibm.com/support/knowledgecenter/en/SSFKSJ_7.1.0/com.ibm.mq.javadoc.doc/WMQJMSClasses/com/ibm/mq/jms/MQConnectionFactory.html#setConnectionNameList_java.lang.String_

Reply:
> Did you try before working on this PR?

I attempted an integration test and setup of multi-instance queue managers but gave up due to my lack of experience with IBM MQ

Well spotted, just setting `connectionNameList ` might not be enough according to these articles:
- https://www.ibm.com/support/knowledgecenter/SSFKSJ_8.0.0/com.ibm.mq.dev.doc/q119450_.htm
- https://www.ibm.com/support/knowledgecenter/SSFKSJ_8.0.0/com.ibm.mq.dev.doc/q119440_.htm

> What if both host and port pair and connectionNameList are set?

This seems to be undefined. Do you agree?

Reply:
The following two quotes are both from the same [IBM page](https://www.ibm.com/support/knowledgecenter/en/SSAW57_9.0.0/com.ibm.websphere.nd.multiplatform.doc/ae/umj_pasm.html):
> Connection name lists can be used to connect to a single queue manager or to a multi-instance queue manager

> This property (connection name list) must only be used to allow connection to a multi-instance queue manager. It must not be used to allow connections to non-multi-instance queue managers as that can result in transaction integrity issues.

I think these quotes contradict each other regarding whether connection name list should be used for single queue manager (broker), but it would be safest to not use it

Furthermore it says:
> If 'Enter host and port information in the form of separate host and port values' is selected, the connection name list property cannot be used and the following properties can be used:

Reply:
I will try to conduct a test next week

Reply:
@ijokarumawak 
Unfortunately the test will be delayed a few weeks due to busyness and easter vacation

Reply:
@vkcelik I understand. Happy holidays :)